test/matcher.cpp

```diff
@@ -113,132 +226,420 @@ TEST_CASE("Matcher throws if regex passed to `=~` or `!~` is invalid",
 	MatcherMockMatchable mock({{"AAAA", "12345"}});
 	Matcher m;
 
-	m.parse("AAAA =~ \"[[\"");
+	REQUIRE(m.parse("AAAA =~ \"[[\""));
 	REQUIRE_THROWS_AS(m.matches(&mock), MatcherException);
 
-	m.parse("AAAA !~ \"[[\"");
+	REQUIRE(m.parse("AAAA !~ \"[[\""));
 	REQUIRE_THROWS_AS(m.matches(&mock), MatcherException);
 }
 
 TEST_CASE("Operator `!~` checks if field doesn't match given regex",
 	"[Matcher]")
 {
-	MatcherMockMatchable mock({{"AAAA", "12345"}});
 	Matcher m;
 
-	m.parse("AAAA !~ \".\"");
-	REQUIRE_FALSE(m.matches(&mock));
+	SECTION("Works with strings") {
```

Comment:
All tests in this block verify that `!~` returns false when the regular expression matches.
Might be nice to add a test in which the regelar expression does not match, resulting in `!~` returning true.